pkg/controller/imagepulljob/imagepulljob_controller.go

```diff
@@ -42,34 +44,48 @@ import (
 	"k8s.io/klog/v2"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-func init() {
-	flag.IntVar(&concurrentReconciles, "imagepulljob-workers", concurrentReconciles, "Max concurrent workers for ImagePullJob controller.")
-}
-
 var (
 	concurrentReconciles        = 3
+	globalParallelism           = 10
 	controllerKind              = appsv1alpha1.SchemeGroupVersion.WithKind("ImagePullJob")
 	resourceVersionExpectations = expectations.NewResourceVersionExpectation()
+	globalLimiter               globallimiter.GlobalLimiter
+	durationStore               = requeueduration.DurationStore{}
 )
 
+func init() {
+	flag.IntVar(&concurrentReconciles, "imagepulljob-workers", concurrentReconciles, "Max concurrent workers for ImagePullJob controller.")
+	flag.IntVar(&globalParallelism, "imagepulljob-global-parallelism", globalParallelism, "Global parallelism for image pulling jobs.")
+}
+
 const (
 	defaultParallelism = 1
 	minRequeueTime     = time.Second
+	defaultRequeueTime = 10 * time.Second
 )
 
 // Add creates a new ImagePullJob Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
 // and Start it when the Manager is Started.
-func Add(mgr manager.Manager) error {
+func Add(mgr manager.Manager) (err error) {
 	if !utildiscovery.DiscoverGVK(controllerKind) || !utilfeature.DefaultFeatureGate.Enabled(features.KruiseDaemon) ||
 		!utilfeature.DefaultFeatureGate.Enabled(features.ImagePullJobGate) {
 		return nil
 	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.ImagePullJobGlobalParallelism) {
+		globalLimiter, err = initializedGlobalLimiter(20 * time.Minute)
```

Comment:
plz replace 20* min with const params, and comment about the relationship with defaultRequeueTime